web/gds-admin-ui/src/utils/index.js

```diff
@@ -119,13 +119,38 @@ function isValidHttpUrl(string) {
 
 /**
  * 
- * @param {Object} contact contact you want to verify
+ * @param {string} contactType contact you want to verify
  * @param {Object} verifiedContact list of verified contacts
  * @returns boolean
  */
-function isVerifiedContact(contact, verifiedContact) {
-    const verifiedContacts = typeof verifiedContact === 'object' ? Object.values(verifiedContact) : []
-    return verifiedContacts.includes(contact.email)
+function isVerifiedContact(contactType = '', verifiedContact) {
+    const verifiedContactsType = typeof verifiedContact === 'object' ? Object.keys(verifiedContact) : []
+    return verifiedContactsType.includes(contactType.toLowerCase())
+}
+
+function isVerify({ data, type = '', verifiedContact }) {
+    let status;
+
+    const contactTypes = Object.keys(verifiedContact)
+    const emails = Object.values(verifiedContact)
+
+    const isIncluded = contactTypes.includes(type.toLowerCase())
+    const isAlternate = emails.includes(data?.email)
+
+    if (isIncluded) {
+        status = VERIFIED_CONTACT_STATUS.VERIFIED
+    } else if (!isIncluded && !isAlternate) {
+        status = VERIFIED_CONTACT_STATUS.UNVERIFIED
+    } else if (!isIncluded && isAlternate) {
+        status = VERIFIED_CONTACT_STATUS.ALTERNATE_VERIFIED
+    }
+
+    return {
+        status
+    }
+
+    // console.log('[verifiedContacts Email]', verifiedContacts)
+    // return verifiedContacts.includes(contactEmail.toLowerCase())
 }
```

Comment:
The `Array.includes` function is an `O(n)` function, meaning that it has to iterate over all of the values in the array to determine if something is included or not. Although the list of contact types and emails is short (4 things), there is still unnecessary work being done here. Can I recommend the following function that will improve the performance of the code?

```suggestion
function isVerify({ data, type = '', verifiedContact }) {
    // Perform verified check
    if (Object.keys(verifiedContact).includes(type.toLowerCase())) {
        return { status: VERIFIED_CONTACT_STATUS.VERIFIED }
    }
    
    // Perform is alternate check
    if (Object.values(verifiedContact).includes(data?.email)) {
        return { status: VERIFIED_CONTACT_STATUS. ALTERNATE_VERIFIED }
    }

    // Contact is unverified
    return { status: VERIFIED_CONTACT_STATUS.UNVERIFIED }
}
```

A couple of things here - the scoping of the arrays `Object.keys` allow the browser to clean things up when they go out of scope; also the short circuit return prevents extra work from happening. Finally the boolean truth table I think is a bit easier to understand without the `!` statements; it works more as an if/else if/else statement. 

Also, I was wondering why you were returning an object instead of just the status itself?